test/browserbehavior/DefaultBrowserBehavior.test.ts

```diff
@@ -115,48 +124,76 @@ describe('DefaultBrowserBehavior', () => {
 
     it('can detect Safari', () => {
       setUserAgent(SAFARI_USERAGENT);
+      domMockBehavior = new DOMMockBehavior();
+      domMockBehavior.usesUnifiedPlan = true;
+      domMockBehavior.browserName = 'safari';
```

Comment:
Gotcha, I updated to use either `setUserAgent` or `domMockBehavior` in DefaultDeviceController.test. But here using `setUserAgent` only won't work because we need to set `domMockBehavior.isUnifiedPlanSupported = false;` for safari 12 & 11 and ios safari 12.0 & 12.1 in domMockBehavior, so that we can turn off unified plan in DOMMockBuilder. So I use `DOMMockBehavior` instead here. 
Let me know if there is a cleaner way to do that :)

Reply:
Using `DOMMockBehavior` makes sense here. I think we should keep replacing `setUserAgent` with `DOMMockBuilder` later.